File: single-language-analysis/dataset-files/python/test_smell_150.py
# Original URL: https://github.com/wookayin/gpustat/blob/7a3b9644229681a752b560c6d4889929b9ea39d5/gpustat/test_gpustat.py#L575-L599

class TestGPUStat(object):

    @pytest.mark.parametrize("scenario_failing_one_gpu", [
        pynvml.NVMLError_GpuIsLost,
        pynvml.NVMLError_Unknown,
    ], indirect=True)
    def test_smell_150(self, scenario_failing_one_gpu):
        """Test a case where one GPU is failing (see #125)."""
        fp = StringIO()
        gpustats = gpustat.new_query()
        gpustats.print_formatted(fp=fp, show_header=False)
        ret = fp.getvalue()
        print(ret)

        lines = remove_ansi_codes(ret).split('\n')
        message = scenario_failing_one_gpu['expected_message']

        # gpu 2: failing due to unknown error
        line = lines[2]
        assert '[2] ((' + message + '))' in line, str(line)
        assert '99999' not in line
        assert '??°C,  ?? %' in line, str(line)
        assert '?? /    ?? MB' in line, str(line)

        # other gpus should be displayed normally
        assert '[0] GeForce GTX TITAN 0' in lines[0]
        assert '[1] GeForce GTX TITAN 1' in lines[1]

File: single-language-analysis/dataset-files/java/TestSmell150.java
import org.junit.Test;
import static org.junit.Assert.*;

public class TestSmell150 {

    @Test
    public void testNewQueryMockedFailingOneGpu() {
        StringWriter fp = new StringWriter();
        GPUSStats gpustats = GPUSStat.newQuery();
        gpustats.printFormatted(fp, false);
        String ret = fp.toString();
        System.out.println(ret);

        String[] lines = removeAnsiCodes(ret).split("\n");
        String message = scenarioFailingOneGpu.get("expected_message");

        // gpu 2: failing due to unknown error
        String line = lines[2];
        assertTrue(line.contains("[2] ((" + message + "))"), line);
        assertFalse(line.contains("99999"));
        assertTrue(line.contains("??°C,  ?? %"), line);
        assertTrue(line.contains("?? /    ?? MB"), line);

        // other gpus should be displayed normally
        assertTrue(lines[0].contains("[0] GeForce GTX TITAN 0"));
        assertTrue(lines[1].contains("[1] GeForce GTX TITAN 1"));
    }
}